gitlab/pkg/adapter/receive_adapter.go

```diff
@@ -177,11 +176,16 @@ func (ra *gitLabReceiveAdapter) postMessage(payload interface{}, source, eventTy
 	event.SetID(eventID)
 	event.SetType(eventType)
 	event.SetSource(source)
-	event.SetDataContentType(cloudevents.ApplicationJSON)
-	event.SetData(payload)
+	err := event.SetData(cloudevents.ApplicationJSON, payload)
+	if err != nil {
+		return err
```

Comment:
Maybe return a better error context such as `"failed to set event data: %w", err`? From my experience the error message returned by `SetData` is too raw to be meaningful without context (like just "json error blah...")